### DISCUSSION TITLE
Can a vector use a memory pool with a small_node_pool ?

### DISCUSSION THREAD

**Discussion by chrisics** [foonathan/memory#170] [Q&A] [answered]
Hello, 
The following throws when the list is constructed. 
Somewhere a proxy to a std::container is allocated with the allocator passed to it and the proxy has size 16 and 16 is larger than sizeof(Item).   
The goal is to speed up vectors with lots of small items, single threaded. Or should I use another allocator for that?    
Thanks,

```
struct Item
{
    int16_t a = 0;
    int16_t b = 0;
};

using MemoryPoolType = memory::memory_pool<memory::small_node_pool>;
using ListType = memory::vector<Item, MemoryPoolType>;

void test()
{
    MemoryPoolType pool(sizeof(Item), 1024);
    ListType list(pool);
    list.push_back(Item());
}

```

**Comment by foonathan** [comment, marked as answer]
vector allocates arrays of small items, so the size of the item isn't an issue, and `memory_pool` isn't great about arrays. If your vector is large, you won't need a custom allocator - the default one should handle it fine. If your vector is small, you have many of them, and don't mind wasting memory for a while, you can use `memory_stack`.
